source/ed/data.cpp

```diff
@@ -497,20 +497,24 @@ create_shape(const nt::GeographicalCoord& from,
 
     nt::LineString res;
     const auto range = get_smallest_range(nearest_from, nearest_to);
-    res.push_back(from);
     if (range.first < range.second) {
         for (auto it = range.first; it <= range.second; ++it)
             res.push_back(*it);
     } else {
         for (auto it = range.first; it >= range.second; --it)
             res.push_back(*it);
     }
-    res.push_back(to);
 
-    // simplification at about 3m precision
+    // simplification
     nt::LineString simplified;
     boost::geometry::simplify(res, simplified, simplify_tolerance);
 
+    // if the shape is only a point, double it in order
+    // to fake a linestring object
+    if (simplified.size() < 2) {
```

Comment:
`if (simplified.size() == 1)`, i don't know this code very well, but if size()==0, it will crash :)

Reply:
Fixed.